modules/data-entry/src/main/frontend/src/questionnaire/Questionnaire.jsx

```diff
@@ -76,9 +78,16 @@ let Questionnaire = (props) => {
       .catch(handleError);
   };
 
-  if (!data) {
-    fetchData();
-  }
+  // When the questionnaire data is fetched, we inform the QuestionnaireContext
+  useEffect(() => {
+    if (data) {
+	  let vars = [];
+      findQuestions(data, vars);
+      changeQuestionnaireContext(vars);
+    } else {
+      fetchData();
+    }
+  }, [data]);
```

Comment:
I think two separate effects make more sense here:

```suggestion
  // First, fetch the questionnaire data
  useEffect(() => {
    fetchData();
  }, []);

  // When the questionnaire data is fetched, we inform the QuestionnaireContext
  useEffect(() => {
    if (data) {
      let vars = [];
      findQuestions(data, vars);
      changeQuestionnaireContext(vars);
    }
  }, [data]);
```